src/realm/group_writer.hpp

```diff
@@ -182,16 +197,27 @@ class GroupWriter : public _impl::ArrayWriterBase {
 
     void write_array_at(MapWindow* window, ref_type, const char* data, size_t size);
     FreeListElement split_freelist_chunk(FreeListElement, size_t alloc_pos);
+
+    /// Backdate (if possible) any blocks in the freelist belonging to
+    /// a version currently becomming unreachable. The effect of backdating
+    /// is that many blocks can be freed earlier.
+    void backdate();
+
+    /// Debug helper
+    std::vector<FreeSpaceEntry> map_reachable();
 };
 
 
 // Implementation:
 
-inline void GroupWriter::set_versions(uint64_t current, uint64_t read_lock) noexcept
+inline void GroupWriter::set_versions(uint64_t current, uint64_t read_lock, TopRefMap& top_refs,
+                                      int num_unreachable) noexcept
```

Comment:
Should this argument be `bool any_new_unreachables`?

Reply:
yup.